lib/matplotlib/legend_handler.py

```diff
@@ -316,7 +316,7 @@ def get_sizes(self, legend, orig_handle,
             numpoints = self.get_numpoints(legend)
             if numpoints < 4:
                 sizes = [.5 * (size_max + size_min), size_max,
```

Comment:
I also vote for drawing the markers in increasing size.  (This also makes the code simpler; now only numpoints=1 needs to be special-cased.)